test/e2e/e2e_test.go

```diff
@@ -1830,6 +1830,225 @@ var _ = ginkgo.Describe("e2e non-vxlan external gateway through a gateway pod",
+		if ip := net.ParseIP(nodeIP); ip == nil {
+			framework.Failf("Unable to retrieve a valid address from container %s with inspect output of %s", workerNodeInfo.name, nodeIP)
+		}
+		framework.Logf("the pod side node is %s and the source node ip is %s", workerNodeInfo.name, nodeIP)
+		podCIDR, err := getNodePodCIDR(workerNodeInfo.name)
+		if err != nil {
+			framework.Failf("Error retrieving the pod cidr from %s %v", workerNodeInfo.name, err)
+		}
+		framework.Logf("the pod cidr for node %s is %s", workerNodeInfo.name, podCIDR)
+
+		// Add loopback addresses used to validate all traffic is getting drained
+		// through the gateway. OVN will choose an ECMP route based on a 5-tuple hash,
+		// so we need to cycle through multiple dest IPs to be able to hit both gateways
+		for _, containerName := range []string{gwContainer1, gwContainer2} {
+			for lastOctet := 1; lastOctet <= ecmpRetry; lastOctet++ {
+				gwLoPrefix := fmt.Sprintf("%s%d/32", exGWRemoteIpPrefix, lastOctet)
+				// add the loopback addresses to the gateway container
+				_, err = runCommand("docker", "exec", containerName, "ip", "address", "add", gwLoPrefix, "dev", "lo")
+				if err != nil {
+					framework.Failf("failed to add the loopback ip to dev lo on the test container %s: %v", containerName, err)
+				}
+			}
+		}
+
+		// Create the pod that will be used as the source for the connectivity test
+		createGenericPod(f, srcPingPodName, workerNodeInfo.name, f.Namespace.Name, command)
+		// wait for the pod setup to return a valid address
+		err = wait.PollImmediate(retryInterval, retryTimeout, func() (bool, error) {
+			pingSrc = getPodAddress(srcPingPodName, f.Namespace.Name)
+			validIP := net.ParseIP(pingSrc)
+			if validIP == nil {
+				return false, nil
+			}
+			return true, nil
+		})
+		// Fail the test if no address is ever retrieved
+		if err != nil {
+			framework.Failf("Error trying to get the pod IP address")
+		}
+		// add a host route on the gateways for return traffic to the pod
+		_, err = runCommand("docker", "exec", gwContainer1, "ip", "route", "add", pingSrc, "via", nodeIP)
+		if err != nil {
+			framework.Failf("failed to add the pod host route on the test container %s: %v", gwContainer1, err)
+		}
+		_, err = runCommand("docker", "exec", gwContainer2, "ip", "route", "add", pingSrc, "via", nodeIP)
+		if err != nil {
+			framework.Failf("failed to add the pod host route on the test container %s: %v", gwContainer2, err)
+		}
+
+		// Verify the gateways and remote loopback addresses are reachable from the pod.
+		// Iterate checking connectivity to the loopbacks on the gateways until tcpdump see
+		// the traffic or 20 attempts fail. Odds of a false negative here is ~ (1/2)^20
+		ginkgo.By(fmt.Sprintf("Verifying ecmp connectivity to the external gateways by iterating through the prefix %s", exGWRemoteIpPrefix))
+
+		// create two channels that will handle error reporting from the goroutines
+		icmpChan1, icmpChan2 := make(chan error), make(chan error)
```

Comment:
Great point, I originally had `icmpChan := make(chan error, 2)` but couldn't think of a clean way to get both errors, ajoea's comment captures both gateways with separate channels if that works for you. Ty!

Reply:
```go
        errs := []error{}
	for i := 0; i < 2; i++ {
		if err := <-icmpCh; err != nil {
			errs = append(errs, err)
		}
	}
       if len(errs) > 1 {
              framework.Failf...
      }
```

Reply:
Huzzah! Only change I'm making to your suggestion @aojea is `if len(errs) > 0` since there could be just one error. Then printing errs rather than errs[0] etc to avoid any out of range index errors.
```
		errs = []error{}
		for i := 0; i < 2; i++ {
			if err := <-icmpChan; err != nil {
				errs = append(errs, err)
			}
		}
		if len(errs) > 0 {
			framework.Failf("failed to reach the mock gateways: \n%v", errs)
		}
```

That returns both gateway or one gateway errors with `[failed to run "docker exec gw-ecmp-test-container1 timeout 10 tcpdump -c 1 icmp": exit status 124 .... (truncating the rest showing zero packets captured)`. Ty!

Reply:
yeah, >1 was a bug, sorry,